Source/WarpX.cpp

```diff
@@ -790,31 +790,43 @@ WarpX::ReadParameters ()
         queryWithParser(pp_algo, "costs_heuristic_cells_wt", costs_heuristic_cells_wt);
         queryWithParser(pp_algo, "costs_heuristic_particles_wt", costs_heuristic_particles_wt);
 
-        int particle_shape;
-        if (pp_algo.query("particle_shape", particle_shape) == false)
-        {
-            amrex::Abort("\nalgo.particle_shape must be set in the input file:"
-                         "\nplease set algo.particle_shape to 1, 2, or 3");
-        }
-        else
-        {
-            if (particle_shape < 1 || particle_shape > 3)
+        // Parse algo.particle_shape and check that input is acceptable
+        // (do this only if there is at least one particle species)
```

Comment:
I know I'm annoying :sweat_smile::
```suggestion
        // (do this only if there is at least one particle or laser species)
```

Reply:
Thank you Edoardo!